### PR TITLE
Add pki_pin only when needed

### PR DESCRIPTION
**Pull request by stlaz**
If both the pki-tomcat NSS database and its password.conf have been
created, don't try to override the password.conf file.

https://pagure.io/freeipa/issue/6839

### CODE REVIEW

File: ipaserver/install/cainstance.py [hunk 1]
```diff
@@ -541,9 +541,13 @@ def __spawn_instance(self):
         # CA key algorithm
         config.set("CA", "pki_ca_signing_key_algorithm", self.ca_signing_algorithm)
 
-        # generate pin which we know can be used for FIPS NSS database
-        pki_pin = ipautil.ipa_generate_password()
-        config.set("CA", "pki_pin", pki_pin)
+        if not (os.path.exists(paths.PKI_TOMCAT_ALIAS_DIR) and
+                os.path.exists(paths.PKI_TOMCAT_PASSWORD_CONF)):
```
Comment (tiran) [inline ipaserver/install/cainstance.py]: Any reason you are using ```exists``` instead of ```isdir``` and ```isfile```?

Comment (stlaz) [inline ipaserver/install/cainstance.py]: No reason, I can update that, thanks.

**Comment by tomaskrizek** [comment]
Replica installation with CA and KRA seems to work fine now.

**Comment by tomaskrizek** [comment]
master:

* 1aa77fe389e957a652c530ec0456ee05467754b3 Add pki_pin only when needed


ipa-4-5:

* f53c76b1055d4f7b26fc127852a66f942845cbae Add pki_pin only when needed

**Pull request closed without merging**